src/character.cpp

```diff
@@ -484,26 +484,46 @@ VisitResponse Character::visit_items( const std::function<VisitResponse( item& )
     return inv.visit_items( func );
 }
 
-VisitResponse Character::visit_items( const std::function<VisitResponse( const item& )>& func ) const
+VisitResponse Character::visit_items( const std::function<VisitResponse( const item *, const item * )>& func ) const
 {
-    return const_cast<Character *>( this )->visit_items( static_cast<const std::function<VisitResponse(item&)>&>( func ) );
+    return const_cast<Character *>( this )->visit_items( static_cast<const std::function<VisitResponse(item *, item *)>&>( func ) );
+}
+
+item * Character::find_parent( item& it )
+{
+    item *res = nullptr;
+    if( visit_items( [&]( item *node, item *parent ){
+        if( node == &it ) {
+            res = parent;
+            return VisitResponse::ABORT;
+        }
+        return VisitResponse::NEXT;
+    } ) != VisitResponse::ABORT ) {
+        debugmsg( "Tried to find item parent using character who doesn't have the item" );
+    }
+    return res;
```

Comment:
It would be a tiny bit safer if the return value of it was a reference instead. We already check for `item.is_null()` all over the code so it wouldn't be unusual to have this one return a null item (`ret_null`) rather than a pointer.